flips/20220503-interaction-templates.md

```diff
@@ -0,0 +1,446 @@
+{
+    f_type: "InteractionTemplate", // Data Type
+    f_vsn: "1.0.0", // Data Type Version
+    id: "a2b2d73def...aabc5472d2", // Unique ID for the data structure.
+    data: {
+        type: "transaction", // "transaction" || "script"
+        interface: "asadf23234...fas234234", // ID of InteractionTemplateInterface this conforms to.
+        version: "1.23.0", // Semver version of this Interaction
+        messages: {
+            title: {
+                i18n: { // Internationalised (BCP-47) set of human readable messages about the interaction
+                    en-US: "Transfer FLOW",
+                    fr-FR: "FLOW de transfert",
+                    zh-CN: "转移流程",
+                }
+            },
+            description: {
+                i18n: { // Internationalised (BCP-47) set of human readable messages about the interaction
+                    en-US: "Transfer {amount} FLOW to {to}", // Messages might consume arguments.
+                    fr-FR: "Transférez {amount} FLOW à {to}",
+                    zh-CN: "将 {amount} FLOW 转移到 {to}"
+                }
+            }
+        },
+        cadence: // Cadence code this interaction executes.
+        ` 
+        import FungibleToken from 0xFUNGIBLETOKENADDRESS
+        transaction(amount: UFix64, to: Address) {
+            let vault: @FungibleToken.Vault
+            prepare(signer: AuthAccount) {
+                %%self.vault <- signer
+                .borrow<&{FungibleToken.Provider}>(from: /storage/flowTokenVault)!
+                .withdraw(amount: amount)%%ount)
+            
+                self.vault <- FungibleToken.getVault(signer)
+            }
+            execute {
+                getAccount(to)
+                .getCapability(/public/flowTokenReceiver)!
+                .borrow<&{FungibleToken.Receiver}>()!
+                .deposit(from: <-self.vault)
+            }
+        }
+        `,
+        dependencies: {
+            "0xFUNGIBLETOKENADDRESS": { // Network (mainnet || testnet) dependent locations for 0xFUNGIBLETOKENADDRESS contract.
+                "FungibleToken" : {
+                    mainnet: {
+                        address: "0xf233dcee88fe0abe", // Address of the account the contract is located.
+                        fq_address: "A.0xf233dcee88fe0abe.FungibleToken", // Fully qualified contract identifier.
+                        pin: "asdfasdfasdfasdfasdfasdfsadf123123123123", // Unique identifier of the interactions dependency tree.
+                        pin_block_height: 10123123123 // Block height the pin was generated against.
+                    },
+                    testnet: {
+                        address: "0x9a0766d93b6608b7",
+                        fq_address: "A.0x9a0766d93b6608b7.FungibleToken",
+                        pin: "asdfasdfasdfasdfasdfasdfsadf123123123123",
+                        pin_block_height: 10123123123
+                    }
+                }
+            }
+        },
+        arguments: {
```

Comment:
Should arguments be typed? example: amount be `ufix64`. I wouldn't expect the type be parsed from cadence script.